### ISSUE TITLE
YAML attribute inconsistencies

### ISSUE THREAD

**Issue by hvarg** [In-For-Disaster-Analytics/metadata-schema#2]
[Metadata-portal](https://github.com/In-For-Disaster-Analytics/metadata-portal) creates YAMLs using the templates defined in this repository.

There are some inconsistencies between what is required for the YAML file and what objects the API returns.
- :white_check_mark: Required and available
- :x: Required in YAML but missing on API responses.
- :grey_question: There is some problem with the YAML encoding. Check notes

| YAML attribute | Model | Layer | Artifact | Notes |
|--------|--------|--------|--------|------|
| `accessibility` |                  :white_check_mark: |  |                   :white_check_mark: | 
| `attribute_accuracy` |              |                   :grey_question: | :x: |  Accuracy is available as an `array` of objects. Each with `name`, `descrip` and `accuracy`.  This YAML field is an string. How we should encode these attributes?    ********** Note from Paul: this is a string field used to hold a statement giving a general assessment of the overall accuracy or completeness of the attribute table associated with the layer. We are reviewing candidate new names for this field to eliminate confusing this with the unrelated `name`, `descrip` and `accuracy` fields. ********** Note from Greg: I am recommending we change it to table_assessment, which addresses both accuracy and completeness of the table. 
| `attribute_codeset_name` |          |                   :white_check_mark: | :x: | 
| `attribute_codeset_url` |           |                   :white_check_mark: | :x: | 
| `attribute_descrip` |               |                   :grey_question: | :x: |  Sames as `attribute_accuracy`  ********** Note from Paul: echo comment from `attribute_accuracy`
| `attribute_domain_type` |           |                   :white_check_mark: | :x: | 
| `attribute_domain_value_descrip` |  |                   :white_check_mark: | :x: | 
| `attribute_name` |                  |                   :grey_question: | :x: |  Sames as `attribute_accuracy`  ********** Note from Paul: echo comment from `attribute_accuracy`
| `attribute_value_range_max` |       |                   :white_check_mark: | :x: | 
| `attribute_value_range_min` |       |                   :white_check_mark: | :x: | 
| `boundary_type` |                  :white_check_mark: | :white_check_mark: | :white_check_mark: | 
| `boundary_unit_id` |               :white_check_mark: | :white_check_mark: | :white_check_mark: |  Using `boundary.code` instead of `boundary.id`
| `classification_sys` |              |                   :x: |  | 
| `classification_term` |             |                   :x: |  | 
| `completeness` |                   :white_check_mark: | :white_check_mark: |  | 
| `contact_affiliation` |            :white_check_mark: | :white_check_mark: | :white_check_mark: |  Using `contact.agency.name`
| `contact_dept` |                   :x: | :x: | :x: | 
| `contact_email` |                  :white_check_mark: | :white_check_mark: | :white_check_mark: | 
| `contact_name` |                   :white_check_mark: | :white_check_mark: | :white_check_mark: | 
| `contact_phone` |                  :white_check_mark: | :white_check_mark: | :white_check_mark: | 
| `contact_role` |                   :white_check_mark: | :white_check_mark: | :white_check_mark: | 
| `creator_name` |                   :white_check_mark: | :white_check_mark: | :white_check_mark: | 
| `custody_chain` |                  :white_check_mark: | :white_check_mark: | :white_check_mark: | 
| `data_stage` |                     :white_check_mark: | :white_check_mark: | :white_check_mark: | 
| `date_created` |                   :white_check_mark: | :white_check_mark: | :white_check_mark: | 
| `date_last_updated` |              :white_check_mark: | :white_check_mark: | :white_check_mark: | 
| `date_submitted` |                 :white_check_mark: | :white_check_mark: | :white_check_mark: | 
| `description` |                    :white_check_mark: | :white_check_mark: | :white_check_mark: | 
| `distribution_method` |            :white_check_mark: | :white_check_mark: | :white_check_mark: | 
| `distribution_url` |               :white_check_mark: | :white_check_mark: | :white_check_mark: | 
| `distributor_publisher` |          :white_check_mark: | :white_check_mark: | :white_check_mark: | 
| `file_format` |                     |                   :white_check_mark: | :white_check_mark: | 
| `horiz_coord_sys` |                 |                   :white_check_mark: |  | 
| `horiz_datum` |                     |                   :white_check_mark: |  | 
| `keyword_term` |                   :white_check_mark: | :white_check_mark: | :white_check_mark: | 
| `keyword_vocab` |                  :white_check_mark: | :white_check_mark: | :white_check_mark: | 
| `license` |                        :white_check_mark: | :white_check_mark: | :white_check_mark: | 
| `process_step_date` |               |                   :x: | :x: | 
| `process_step_descrip` |            |                   :x: | :x: | 
| `process_step_source` |             |                   :x: | :x: | 
| `model_dim` |                      :x: |  |                    | 
| `model_type` |                     :grey_question: |  |                    |  `model.types` is an `array`, this field is an `String` ********** Note from Paul: this field should accept multiple values chosen from a controlled vocabulary list. Type `array` sounds appropriate.
| `object_version` |                 :white_check_mark: |  |                    | 
| `purpose` |                        :white_check_mark: | :white_check_mark: | :white_check_mark: | 
| `scale` |                          :white_check_mark: | :white_check_mark: |  | 
| `service_endpoint_type` |          :x: | :x: | :x: | 
| `software_name` |                  :white_check_mark: |  |                    | 
| `software_version` |               :white_check_mark: |  |                    | 
| `spatial_accuracy` |               :white_check_mark: | :white_check_mark: |  | 
| `spatial_data_type` |               |                   :white_check_mark: |  | 
| `spatial_extent` |                 :white_check_mark: | :white_check_mark: |  | 
| `time_period_covered` |    :white_check_mark:           |           :white_check_mark:           |                   :white_check_mark: | 
| `title` |                          :white_check_mark: | :white_check_mark: | :white_check_mark: | 
| `update_freq` |                    :white_check_mark: | :white_check_mark: | :white_check_mark: | 
| `use_constraint` |                 :white_check_mark: | :white_check_mark: | :white_check_mark: | 
| `vert_coord_sys` | |                                    :white_check_mark: |  | 

For :x: we should choose if we add the property to the API or we remove the attribute for the YAML template.

**Comment by brentporter** [comment]
Greg - can you get in to look at these - particularly on the artifact stuff there is still some confusion about what is required vs recommended

**Comment by gsmithhart** [comment]
Hey guys. I just wanted to note that **attribute_accuracy** is meant to be a general description (in free text) of the "overall health" of the entire attribute table associated with a layer or artifact, NOT specific to each field in a given attribute table. So not one-for-one related to **attribute description** and **attribute name** which ARE required for each field in an attribute table. This is one thing that the folks over at Lamar stumbled over as well. I'm actually now wanting to rename that element so it does not have the prefix of "attribute_". Maybe something like overall_table_accuracy. I realized when working with Lamar that this was going to be confusing.

**Comment by paul-csr** [comment]
I agree with the re-name, so it's not conflated with the attribute domain elements. Or shorter "table_accuracy" would work.
________________________________

**Comment by brentporter** [comment]
@urban011 I want to loop you into this to see the discussion

**Comment by gsmithhart** [comment]
Hey guys. Sorry for the delay on this. I created an Excel based on an extract from our spreadsheet that feeds the GitHUB page. I removed a few columns to reduce cluster and focus on the question at hand. I then added columns for all of the metadata elements and then populated cells with X's indicating which are required and each tier, essentially creating a matrix.  All required metadata elements are shaded green to easily differentiate from those that are not absolutely required. It's possible we may need to adjust some of these, but this is the way it is now in the master spreadsheet. 

[Required Metadata Matrix.xlsx](https://github.com/In-For-Disaster-Analytics/metadata-schema/files/13640710/Required.Metadata.Matrix.xlsx)
